common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -1233,7 +1233,12 @@ def submit_problem(self, data, override_time=False):
             # without a stack trace
             else:
                 # Translators: {msg} will be replaced with a problem's error message.
-                msg = inst.message
+                try:
+                    # only return the error value of the exception
+                    message = inst.message.split("\\n")[-2].split(": ", 1)[1]
+                    msg = cgi.escape(message)
+                except IndexError:
+                    msg = inst.message
```

Comment:
Yeah. I switched to inst.message because it seemed clearer, but for reasons I don't understand it doesn't seem as reliable as `inst.args[0]`. I've switched it back.